### ISSUE TITLE
`S404` false positive in `TYPE_CHECKING` block

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#9528]
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from subprocess import run
```
https://play.ruff.rs/d722ef1d-09fd-46c7-82c5-23d85498c6ae

since it's only used when type-checking, there's no security risk so this rule should not be triggered

**Comment by KotlinIsland** [comment]
Inb4 my voln repro does `typing.TYPE_CHECKING = True`

**Comment by charliermarsh** [comment]
Torn on this because it's typically a false positive but it's also a security-oriented rule so it feels strange to be lax on it.

**Comment by eli-schwartz** [comment]
I would assume that a typing-only import of the subprocess module indicates that the code in question either returns or accepts subprocess module types and is therefore "insecure" for this reason.

The fact that it was only detected via the type signature rather than via a direct import seems immaterial. The currently reported violation allows detecting things like:

```python
import typing as T

from fooutils import Popen_unicode
from myapp import load_settings

if T.TYPE_CHECKING:
    from subprocess import DEVNULL, PIPE

cfg = load_settings()
r, out, err = Popen_unicode(cfg['command'], stdout=PIPE, stderr=DEVNULL)
print(out)
```
